android/app/src/main/java/com/magnet/webview/MagnetWebViewManager.java

```diff
@@ -0,0 +1,122 @@
+package com.magnet.webview;
+
+import javax.annotation.Nullable;
+
+import android.os.Build;
+import android.webkit.WebView;
+
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.common.build.ReactBuildConfig;
+import com.facebook.react.uimanager.SimpleViewManager;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MagnetWebViewManager extends SimpleViewManager<WebView> {
+    private static final String REACT_CLASS = "MagnetWebView";
+    private static final String HTML_ENCODING = "UTF-8";
+    private static final String HTML_MIME_TYPE = "text/html; charset=utf-8";
+    private static final String HTTP_METHOD_POST = "POST";
+
+    // Use `webView.loadUrl("about:blank")` to reliably reset the view
+    // state and release page resources (including any running JavaScript).
+    private static final String BLANK_URL = "about:blank";
+
+    @Override
+    public String getName() {
+        return REACT_CLASS;
+    }
+
+    @Override
+    protected WebView createViewInstance(ThemedReactContext reactContext) {
+        MagnetWebView webView = new MagnetWebView(reactContext);
+        reactContext.addLifecycleEventListener(webView);
+
+        if (ReactBuildConfig.DEBUG && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            WebView.setWebContentsDebuggingEnabled(true);
+        }
+
+        return webView;
+    }
+
+    @Override
+    protected void addEventEmitters(ThemedReactContext reactContext, WebView view) {
+        // Do not register default touch emitter and let WebView implementation handle touches
+        view.setWebViewClient(new MagnetWebViewClient());
+    }
+
+    @Override
+    public @Nullable Map getExportedCustomDirectEventTypeConstants() {
+        return MapBuilder.of(
+                "magnetWebViewLoaded",
+                MapBuilder.of("registrationName", "onMagnetWebViewLoaded")
+        );
+    }
+
+    @Override
+    public void onDropViewInstance(WebView webView) {
+        ((ThemedReactContext) webView.getContext()).removeLifecycleEventListener((MagnetWebView) webView);
+        ((MagnetWebView) webView).cleanupCallbacksAndDestroy();
+    }
+
+    @ReactProp(name = "source")
+    public void setSource(WebView view, @Nullable ReadableMap source) {
+        if (source != null) {
+            if (source.hasKey("html")) {
+                String html = source.getString("html");
+
+                if (source.hasKey("baseUrl")) {
+                    view.loadDataWithBaseURL(
+                            source.getString("baseUrl"), html, HTML_MIME_TYPE, HTML_ENCODING, null);
+                } else {
+                    view.loadData(html, HTML_MIME_TYPE, HTML_ENCODING);
```

Comment:
Question (don't have the answer), should we allow to load html source code in our views? Any use case?